### PR TITLE
ci: update build setup scripts to Qt 5.15.2

### PR DESCRIPTION
**Pull request by jakubgs**
This needs to be synced with a Squish upgrade to match the QT version.

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
LGTM

**Comment by jakubgs** [comment]
Looks like I accidentally installed Node.js `19.x` on one of the MacOS i7 hosts and got this:
```
npm ERR! make[1]: warning: jobserver unavailable: using -j1.  Add `+' to parent make rule.
npm ERR! In file included from ../src/volume.cc:2:
npm ERR! In file included from ../../nan/nan.h:178:
npm ERR! ../../nan/nan_callbacks.h:55:23: error: no member named 'AccessorSignature' in namespace 'v8'
npm ERR! typedef v8::Local<v8::AccessorSignature> Sig;
npm ERR!                   ~~~~^
npm ERR! In file included from ../src/volume.cc:2:
npm ERR! ../../nan/nan.h:2544:8: error: no matching member function for call to 'SetAccessor'
npm ERR!   tpl->SetAccessor(
npm ERR!   ~~~~~^~~~~~~~~~~
npm ERR! /Users/jenkins/Library/Caches/node-gyp/19.5.0/include/node/v8-template.h:814:8: note: candidate function not viable: no known conversion from 'imp::Sig' (aka 'int') to 'v8::SideEffectType' for 7th argument
npm ERR!   void SetAccessor(
npm ERR!        ^
npm ERR! /Users/jenkins/Library/Caches/node-gyp/19.5.0/include/node/v8-template.h:807:8: note: candidate function not viable: no known conversion from 'imp::NativeGetter' (aka 'void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value> &)') to 'v8::AccessorGetterCallback' (aka 'void (*)(Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') for 2nd argument
npm ERR!   void SetAccessor(
npm ERR!        ^
npm ERR! 2 errors generated.
```
Rolling back to Node.js `18.x` or `16.x` fixes it tho.

**Comment by jakubgs** [comment]
Here's some notes on dev environment setup but they are quite old:
https://hackmd.io/OtFjDnH6QtOl3K65fKKEGg#Building-Status-Desktop

This kind of stuff really should be in the repo and kept up-to-date.

**Comment by jakubgs** [comment]
I'm getting some weird build failures from `e2e` build:
```
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickFlickablePrivate::flick(QQuickFlickablePrivate::AxisData&, double, double, double, void (*)(void*), double)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlObjectCreator::finalizePopulateDeferred()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::syncSyncView()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlDelegateModel::drainReusableItemsPool(int)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::fixup(QQuickFlickablePrivate::AxisData&, double, double)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlObjectCreator::beginPopulateDeferred(QQmlContextData*)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::~QQuickTableViewPrivate()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `qt_version_tag@Qt_5.15'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlObjectCreator::populateDeferredBinding(QQmlProperty const&, int, QV4::CompiledData::Binding const*)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `non-virtual thunk to QQuickFlickablePrivate::itemGeometryChanged(QQuickItem*, QQuickGeometryChange, QRectF const&)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::syncWithPendingChanges()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::loadedTableItem(QPoint const&) const@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::setModelImpl(QVariant const&)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickWindowPrivate::dragOverThreshold(double, Qt::Axis, QTouchEvent::TouchPoint const*, int)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlComponentPrivate::setInitialProperties(QV4::ExecutionEngine*, QV4::QmlContext*, QV4::Value const&, QV4::Value const&, RequiredProperties&, QObject*)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickFlickablePrivate::addPointerHandler(QQuickPointerHandler*)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickWindow::tabletEvent(QTabletEvent*)@Qt_5'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::modelImpl() const@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::syncDelegate()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlDelegateModel::poolSize()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickFlickablePrivate::itemGeometryChanged(QQuickItem*, QQuickGeometryChange, QRectF const&)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::syncModel()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQmlDelegateModel::release(QObject*, QQmlInstanceModel::ReusableFlag)@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `typeinfo for QQuickTableViewPrivate@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::updatePolish()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QQuickTableViewPrivate::QQuickTableViewPrivate()@Qt_5_PRIVATE_API'
/usr/bin/ld: /opt/qt/5.15.2/gcc_64/lib/libQt5QuickTemplates2.so.5: undefined reference to `QRegularExpression::anchoredPattern(QStringView)@Qt_5'
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: 'g++  @nim_status_client_linkerArgs.txt'
```
Must be some difference between new and old Qt installations.

**Comment by jakubgs** [comment]
And there we have it:
```
admin@linux-06.he-eu-hel1.ci.devel:~ % sudo su - jenkins
jenkins@linux-06:~$ qmake --version
QMake version 3.1
Using Qt version 5.14.2 in /opt/qt/5.14.2/gcc_64/lib
```
Jenkins user has wrong `qmake` version in `PATH`.

File: scripts/macos_build_setup.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  echo "Install build dependencies"
+  brew install cmake pkg-config libtool jq node@18
+}
+
+function install_qt {
+  echo "Install QT"
+  brew install python@3.10
+  pip3 install -U pip
+  pip3 install aqtinstall
+  aqt install-qt mac desktop ${QT_VERSION} clang_64 -m qtwebengine qtlottie -O ${QT_INSTALL_DIR}
```
Comment (caybro) [inline scripts/macos_build_setup.sh:26]: We don't use lottie any longer

File: scripts/ubuntu_build_setup.sh [hunk 2]
```diff
@@ -40,19 +45,35 @@ function install_qt {
   apt install -y python3-pip
   pip install -U pip
   pip install aqtinstall
-  aqt install-qt linux desktop 5.14.2 gcc_64 -m qtwebengine qtlottie -O /opt/qt
+  aqt install-qt linux desktop ${QT_VERSION} gcc_64 -m qtwebengine qtlottie -O ${QT_INSTALL_DIR}
```
Comment (caybro) [inline scripts/ubuntu_build_setup.sh:48]: Neither here

**Review by caybro — APPROVED** [review]
Otherwise lgtm

**Comment by jakubgs** [comment]
I'vei installed the new Squish plugin version on all Linux CI hosts:
```
 > a ci-slave-linux -o -a 'ls /opt | grep squish-runner-7.1.0'
linux-01.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-02.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-03.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-04.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-05.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-06.he-eu-hel1.ci.devel | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
linux-01.he-eu-hel1.ci.release | CHANGED | rc=0 | (stdout) squish-runner-7.1.0
```
And configured the installation in `` on Jenkins master:

image

**Comment by jakubgs** [comment]
Nice, after updating `squishPackageName` to the new version:
https://github.com/status-im/status-desktop/blob/8a8430ebfd8f380132c7e9234c0ce762503d6496/ci/Jenkinsfile.e2e#L142
The `e2e` build stopped looking like this:

image

And now looks like this:

image

**Comment by jakubgs** [comment]
Here's full set of changes for Qt scripts, Squish runner, and Squish license server upgrades:

- https://github.com/status-im/infra-ci/commit/2b581fe2 - status-desktop-setup: upgrade to install Qt 5.15.2
- https://github.com/status-im/infra-ci/commit/0f5a2313 - status-desktop-setup: add use of script for MacOS
- https://github.com/status-im/infra-ci/commit/a5f45579 - squish-runner: upgrade from 6.7.2 to 7.1.0
- https://github.com/status-im/infra-ci/commit/627fea05 - squish-license-server: upgrade from 6.7.2 to 7.1.0

One thing worth noting is that the old Squish license server worked fine with the new Squish runner.

**Pull request merged**